js/livecoin.js

```diff
@@ -71,21 +71,46 @@ module.exports = class livecoin extends Exchange {
 
     async fetchMarkets () {
         let markets = await this.publicGetExchangeTicker ();
+        let commissionInfo = await this.privateGetExchangeCommissionCommonInfo ();
```

Comment:
Unfortunately, we can't do this. Can't call multiple private methods from a public method. Have to rethink this and split into multiple methods.

Reply:
@kroitor there's no other way to get maker/taker fees on *livecoin*, 'cause it depends on trade volume.

Reply:
@orkenstein then we should not do that in fetchMarkets?

Reply:
@kroitor agree, it should be removed from public part. 
What method?

Reply:
@orkenstein see Poloniex `fetchFees()` – that one is private. However, it lacks parsing and populating the markets with fees. But the method name is `fetchFees()`.

Reply:
@kroitor what if call `this.privateGetExchangeCommissionCommonInfo` only if credentials provided?

Reply:
@orkenstein it will mostly be confusing to users if it behaves this or that way. We believe, it's better to remain explicit, so we would add a separate private method for the job.

Reply:
My point: there's a default value (0.18%), so if you're not logged in - it's default.
But when we access API with credentials we can refine results.

Reply:
@orkenstein at the same time we want to control private requests to avoid nonce conflicts and rate-limiting issues, so we can't really decide for the user when to hit the internet. In terms of library design it it still much better to have a separate method for that.

Reply:
@kroitor, yes, makes sense.

Reply:
@orkenstein think this way, we can always make up a wrapper method and/or fetch conditionally depending on the presence of API credentials, but we should first split that whole thing into smallest possible pieces, and then recombine upwards from there. So, it turns out, we need separate methods anyway. So, let's have small methods in the lib, and then let the user do his wrappers if needed. And then, when we have all the lower tier polished, we will add wrappers of our own, more layers of abstraction, and so on... For now we should stick with robustness.

Reply:
@kroitor 
There's no unified API for fees?

Reply:
Trading fees are mostly unified, however, you still have to pay attention to that in userland. Deposit and withdrawal fees are in process of unification right now (fetchCurrencies).

Reply:
@kroitor 
>Trading fees are mostly unified

I only see `fetchFees()` in **Poloniex**. What method should I implement to use `this.privateGetExchangeCommissionCommonInfo()`?

Reply:
`fetchFees()` for Livecoin? )

Reply:
@kroitor, I've expected to see *unified solution* for fees across all exchanges.

Reply:
@orkenstein from a library that is under development?)) Just kidding, of course, you would! But don't you think `fetchFees()` is the most suitable name for the job? This could be the start of unification, right? Otherwise we will be expecting it forever. When working with this library you have to realize: if something is not there, it's because we didn't have enough time to make it, just the way it is. In general, if something is missing, expecting it is less constructive than making it ;) The whole library should contain approximately 2000 unified methods, we have implemented (unified) more than 60% of them by now. What I'm saying is: we would be happy if you submit a couple of methods )

Reply:
@kroitor, next time just tell me: it's under development :D